Rebrand profile page

el-mapache opened the pull request:
**Why**: Better readability and a mobile first design


Screenshots:

<img width="851" alt="screen shot 2017-04-10 at 4 44 16 pm" src="https://cloud.githubusercontent.com/assets/1421848/24881855/24f0073e-1e0d-11e7-8f49-ed02f1766836.png">

<img width="752" alt="screen shot 2017-04-10 at 4 44 25 pm" src="https://cloud.githubusercontent.com/assets/1421848/24881867/2a076d52-1e0d-11e7-93d8-f0752d74fb96.png">

<img width="497" alt="screen shot 2017-04-10 at 4 44 47 pm" src="https://cloud.githubusercontent.com/assets/1421848/24881881/32a7cf4c-1e0d-11e7-9dc4-b8c42cf21676.png">

<img width="192" alt="screen shot 2017-04-10 at 4 45 34 pm" src="https://cloud.githubusercontent.com/assets/1421848/24881884/38095bc2-1e0d-11e7-87b0-e38eda84e804.png">

rtwell commented:
hey @el-mapache this is looking fantastic! 

a couple things i spotted that should be addressed before we push this through:

1) we're missing the **Account information** header on mobile

2) the blue in the box-headers should be the same blue as the _Welcome…_ background

el-mapache commented:
@rtwell I changed the blue color, and added back the 'Your account' header on mobile!

<img width="444" alt="screen shot 2017-04-11 at 5 03 01 pm" src="https://cloud.githubusercontent.com/assets/1421848/24930764/b636eb88-1ed8-11e7-9af2-2c96e858cc80.png">

el-mapache commented:
Looking at it now, it seems like we will need a different header entirely for a non-verified account, since the spacing currently takes the presence of the verified badge as a given.

rtwell commented:
This is how I imagined it:
![image](https://cloud.githubusercontent.com/assets/12564977/24931026/44bc5274-1ed1-11e7-9f9f-fc3fd79242db.png)

el-mapache commented:
@rtwell I just left a comment above re: the spacing, but the email currently only displays on a separate line in the case of a long email — like `the homerjsimpson@hotmail.com` that you have in the comps. 

Should the email always display on a separate line, regardless of length?  I'm assuming the name is going to always appear inline unless it is a very long first name.

rtwell commented:
Ah sorry Adam—i should have specified: Yes, i imagined the email addy, in
bold, on its own line, on mobile.

On Tue, Apr 11, 2017 at 4:16 PM, Adam B <notifications@github.com> wrote:

> @rtwell <https://github.com/rtwell> I just left a comment above re: the
> spacing, but the email currently only displays on a separate line in the
> case of a long email — like the homerjsimpson@hotmail.com that you have
> in the comps.
>
> Should the email always display on a separate line, regardless of length?
> I'm assuming the name is going to always appear inline unless it is a very
> long first name.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/18F/identity-idp/pull/1352#issuecomment-293402386>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL-58QyB2WkYXxdpaexoO8QVt09CMUUsks5ru-2ZgaJpZM4M5TP->
> .
>

rtwell commented:
I realize that is different than the LOA3 instance 😱  but since there is no verified mark, it felt strange sitting by itself. Breaking it onto two lines will deal with the majority of lengthy government email addresses (and avoid truncating them)

el-mapache commented:
@rtwell understood, will do!

el-mapache commented:
@rtwell Here is the header for an unverified user:

![screen shot 2017-04-12 at 9 00 12 am](https://cloud.githubusercontent.com/assets/1421848/24958841/c5a3ac7e-1f5e-11e7-8fd1-9c7ea621ee4a.png)


and a verified one:

![screen shot 2017-04-12 at 9 06 15 am](https://cloud.githubusercontent.com/assets/1421848/24958999/486fa23e-1f5f-11e7-9da6-2d061e2fdfd0.png)

hursey013 commented:
I noticed on LOA1 the "Profile information" heading is still visible, is that expected?
![screen shot 2017-04-13 at 3 30 12 pm](https://cloud.githubusercontent.com/assets/1178494/25020992/81aa4a0e-205e-11e7-9a0f-d44a4681fe14.png)

el-mapache commented:
No, that looks like a merge conflict issue, it wasn't there before 😞

hursey013 commented on app/views/profile/_header.html.slim:
```diff
@@ -1,11 +1,14 @@
-.clearfix.pt2.pb3.sm-py0.bg-light-blue.sm-bg-none
+.clearfix.pt2.pb2.sm-py0.mb2.bg-light-blue.sm-bg-none
```
Small thing, `.pt2.pb2` can be condensed to `.py2`

el-mapache commented:
Looking at this again there are now a million problems with the visuals

hursey013 commented on app/views/profile/_profile_item.html.slim:
```diff
@@ -0,0 +1,11 @@
+.p2.clearfix
```
A few unnecessary grid classes:
```
.p2.clearfix
  .col.col-8.clearfix
    .sm-col.sm-col-6.bold
      = name
    .sm-col.sm-col-6
      = content if local_assigns.key? :content
  .col.col-4.right-align
    - if local_assigns.key? :path
      = render action, path: path, name: name
    - else
      = render action
```

hursey013 commented on app/views/profile/index.html.slim:
```diff
@@ -14,50 +13,59 @@ h1.hide = t('titles.profile')
+    content: current_user.email,
+    path: manage_email_path,
+    action: @view_model.edit_action_partial
+  .border-bottom.border-blue-light
```
Could this line be added to the containing `.p2.clearfix` of `_profile_item.html.slim` instead of having the extra div?

el-mapache replied:
I don't think so, because we want to be able to define when we have a bottom border (e.g. not on the last item)

el-mapache replied:
I went with a border-top to remove this duplication.

hursey013 commented on app/views/shared/_nav_auth.html.slim:
```diff
@@ -1,13 +1,14 @@
   .container.px2.sm-py3.pt2.pb0
     .clearfix
-      .sm-col.mb1.sm-m0.col.col-8
+      .sm-col.mb1.sm-m0.col.col-7
```
`.sm-col` not necessary here

hursey013 commented on app/views/shared/_nav_auth.html.slim:
```diff
@@ -1,13 +1,14 @@
             class: 'align-bottom',
             width: 170), root_path
-      .sm-col-right.right-align.col.col-4
+      .sm-col-right.right-align.col.col-5
```
`.sm-col-right` not necessary here

hursey013 commented on app/assets/stylesheets/components/_background.scss:
```diff
@@ -0,0 +1 @@
+.bg-gray-lighter { background-color: $gray-lighter; }
```
We have `bg-light-blue` and `sm-bg-light-blue` in `util.scss` so we might want to consolidate our custom background colors.

el-mapache replied:
Would you prefer to have those classes in util or in a background.scss file?

hursey013 replied:
I'm guessing we may continue to add additional variations so I think having them all in `background.scss` makes sense.

hursey013 commented on vendor/assets/stylesheets/basscss/_border-colors.scss:
```diff
@@ -11,6 +11,8 @@
 
 .border-gray   { border-color: $gray }
 
+.border-blue-lighter { border-color: $blue-lighter; }
```
I think we should move this into our own `border.scss` vs the vendor stylesheet just in case we ever update Basscss.

el-mapache replied:
yeah absolutely, I thought I moved this out already!

hursey013 commented:
This is looking awesome, just noticed a few super small things when running locally then should be good to go.

LOA3 on mobile, Profile information looks like the rounded corners are still present and needs a bottom border: 
<img width="191" alt="screen shot 2017-04-14 at 10 13 31 am" src="https://cloud.githubusercontent.com/assets/1178494/25045828/20e1839c-20fc-11e7-8d6b-95f8dfb85285.png">

Bottom border on this gray div ends just a tad bit short:
<img width="198" alt="screen shot 2017-04-14 at 10 13 54 am" src="https://cloud.githubusercontent.com/assets/1178494/25045970/e0dd81dc-20fc-11e7-94da-aa7e08bf3876.png">

I think the values in login information and 2fa section need to move one column to the right so all of the values are in line:
<img width="736" alt="screen shot 2017-04-14 at 10 25 39 am" src="https://cloud.githubusercontent.com/assets/1178494/25046002/09c6345e-20fd-11e7-91e2-62add32ee284.png">

hursey013 commented on app/views/profile/_pii.html.slim:
```diff
@@ -1,22 +1,41 @@
-.mt3.sm-mt2.mb4
-  .py-12p.border-top
+.border.border-blue-lighter.rounded-md.mb3
+  .bg-light-blue.py2.px2.h6.caps.clearfix
```
Switch the `.py2` to `.py1` to match the padding of the other panel headings.

el-mapache commented:
@hursey013 cleaned up!

<img width="329" alt="screen shot 2017-04-14 at 11 13 07 am" src="https://cloud.githubusercontent.com/assets/1421848/25047203/5eb7c8d2-2103-11e7-9dd5-b0c5b6e6c70e.png">

<img width="546" alt="screen shot 2017-04-14 at 11 12 56 am" src="https://cloud.githubusercontent.com/assets/1421848/25047207/622bd7e2-2103-11e7-8664-49fe060e0d0d.png">

hursey013 commented on app/assets/stylesheets/components/_border.scss:
```diff
@@ -10,6 +10,9 @@
 .rounded-xl { border-radius: $border-radius-xl; }
 .rounded-xxl { border-radius: $border-radius-xxl; }
 
+.border-blue-lighter { border-color: $blue-lighter; }
```
This doesn't look like it's being used, and after talking it through with @rtwell this can also just fall back to using the default `$border-color`.
